### PR TITLE
Clear the activation key if exists in the DB

### PR DESCRIPTION
**Pull request by Rajinsharwar**
Clear the activation key if exists in the DB during login.

Trac ticket: https://core.trac.wordpress.org/ticket/58901

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/user.php [hunk 1]
```diff
@@ -110,6 +110,23 @@ function wp_signon( $credentials = array(), $secure_cookie = '' ) {
 	}
 
 	wp_set_auth_cookie( $user->ID, $credentials['remember'], $secure_cookie );
+
+	// Flush `user_activation_key` if exists after successful login.
+	global $wpdb;
```
Comment (mukeshpanchal27) [inline src/wp-includes/user.php:117]: Add global doc for new global.

Comment (Rajinsharwar) [reply src/wp-includes/user.php:117]: Done, thanks for the catch!

File: src/wp-includes/user.php [hunk 2]
```diff
@@ -110,6 +110,26 @@ function wp_signon( $credentials = array(), $secure_cookie = '' ) {
...
 	wp_set_auth_cookie( $user->ID, $credentials['remember'], $secure_cookie );
+
+	/**
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 */
+	global $wpdb;
+
+	// Flush `user_activation_key` if exists after successful login.
+	if ( ! empty( $user->user_activation_key ) ) {
+		$wpdb->update(
```
Comment (Rahmon) [inline src/wp-includes/user.php:121]: Hi @Rajinsharwar 

I would just suggest passing the parameters `$format` and `$where_format` to reinforce the expected formats.

```php
$wpdb->update(
	$wpdb->users,
	array( 'user_activation_key' => '', ),
	array( 'ID' => $user->ID ),
	array( '%s' ),
	array( '%d' )
);
```

Other than that, looks good to me :raised_hands:

Comment (Rajinsharwar) [reply src/wp-includes/user.php:121]: Thanks for your suggestion! Added :)

**Comment by audrasjb** [comment]
COmmitted in https://core.trac.wordpress.org/changeset/58333

**Pull request closed without merging**